qos/evm/response_generic.go

```diff
@@ -3,46 +3,88 @@ package evm
 import (
 	"encoding/json"
 
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	qosobservations "github.com/buildwithgrove/path/observation/qos"
 	"github.com/buildwithgrove/path/qos/jsonrpc"
 )
 
+const (
+	// errCodeUnmarshalling is set as the JSONRPC response's error code if the endpoint returns a malformed response
+	errCodeUnmarshalling = -32600
+	// errMsgUnmarshalling is the generic message returned to the user if the endpoint returns a malformed response.
+	errMsgUnmarshalling = "the response returned by the endpoint is not a valid JSONRPC response"
+
+	// errDataFieldRawBytes is the key of the entry in the JSONRPC error response's "data" map which holds the endpoint's original response.
+	errDataFieldRawBytes = "endpoint_response"
+
+	// errDataFieldUnmarshallingErr is the key of the entry in the JSONRPC error response's "data" map which holds the unmarshalling error.
+	errDataFieldUnmarshallingErr = "unmarshalling_error"
+)
+
 // TODO_UPNEXT(@adshmh): implement the generic jsonrpc response
 // (with the scope limited to an EVM-based blockchain)
 // responseGeneric captures the fields expected in response to any request on an
 // EVM-based blockchain. It is intended to be used when no validation/observation
 // is applicable to the corresponding request's JSONRPC method.
 // i.e. when there are no unmarshallers/structs matching the method specified by the request.
 type responseGeneric struct {
-	ID      jsonrpc.ID      `json:"id"`
-	JSONRPC jsonrpc.Version `json:"jsonrpc"`
-
-	rawBytes         []byte
-	unmarshallingErr error
+	jsonrpc.Response
+	Logger polylog.Logger
 }
 
-func (r responseGeneric) GetObservation() (observation, bool) {
-	return observation{}, false
+// GetObservation returns an observation that is NOT used in validating endpoints.
```

Comment:
#PUC why

Reply:
The next line of the comment provides some details, please reopen if that needs an update.